packages/discovery-provider/src/tasks/entity_manager/utils.py

```diff
@@ -444,16 +452,47 @@ def validate_signer(params: ManageEntityParameters):
         is_signer_authorized = grant_key in params.existing_records["Grant"]
         if is_signer_authorized:
             grant = params.existing_records["Grant"][grant_key]
-            developer_app = params.existing_records["DeveloperApp"][signer]
-            if (not developer_app) or (developer_app.is_delete) or (grant.is_revoked):
+            developer_app = (
+                None
+                if not params.existing_records["DeveloperApp"]
+                else params.existing_records["DeveloperApp"].get(signer, None)
+            )  # applicable if user-to-app grant
+            user_grantee = (
+                None
+                if not params.existing_records["UserWallet"]
+                else params.existing_records["UserWallet"].get(signer, None)
+            )  # applicable if user-to-user grant, e.g. manager mode
+            is_valid_developer_app = (
+                not not developer_app and not developer_app.is_delete
+            )
+
+            is_valid_user_grantee = (
+                not not user_grantee and not user_grantee.is_deactivated
```

Comment:
`not not`? Is this checking that the `user_grantee` isn't `None`? If so, can we do a more explicit check here? This is a bit hard to read :-)

Reply:
ok confirmed that making this sensical results inexplicably results in a mypy error. i added parens to make the not not more clear.